scholia/rss.py

```diff
@@ -0,0 +1,212 @@
+"""rss
+
+Usage:
+  scholia.rss author-latest <q>
+
+
+Description:
+  Interface to the Wikidata API and its bibliographic data.
+
+Examples:
+  $ python -m scholia.rss author-latest Q27061849
+  <rss />
+
+"""
+
+
+from __future__ import print_function
+
+import requests
+from dateutil.parser import parse as parse_datetime
+from email.utils import formatdate
+import time
+# from uuid import uuid3, NAMESPACE_URL
+
+SPARQL_QUERY = """
+PREFIX bd: <http://www.bigdata.com/rdf#>
+SELECT ?work ?workLabel ?date
+WITH {{
+  SELECT
+    (MIN(?dates) AS ?date) ?work
+  WHERE {{
+    ?work wdt:P50 wd:{q} .
+    ?work wdt:P577 ?datetimes .
+    BIND(xsd:date(?datetimes) AS ?dates)
+  }}
+  GROUP BY ?work ?workLabel
+  ORDER BY DESC(?date)
+  LIMIT 10
+}} AS %content {{
+  INCLUDE %content
+  SERVICE wikibase:label {{ bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }}
+}} ORDER BY DESC(?date)
+"""
+
+VENUE_SPARQL_QUERY = """
+PREFIX bd: <http://www.bigdata.com/rdf#>
+SELECT ?work ?workLabel ?date
+WITH {{
+  SELECT
+    (MIN(?dates) AS ?date) ?work
+  WHERE {{
+    ?work wdt:P1433 wd:{q} .
+    ?work wdt:P577 ?datetimes .
+    BIND(xsd:date(?datetimes) AS ?dates)
+  }}
+  GROUP BY ?work ?workLabel
+  ORDER BY DESC(?date)
+  LIMIT 10
+}} AS %content {{
+  INCLUDE %content
+  SERVICE wikibase:label {{ bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }}
+}} ORDER BY DESC(?date)
+"""
+
+TOPIC_SPARQL_QUERY = """
+PREFIX bd: <http://www.bigdata.com/rdf#>
+SELECT ?work ?workLabel ?date
+WITH {{
+  SELECT
+    (MIN(?dates) AS ?date) ?work
+  WHERE {{
+    ?work wdt:P921 / (wdt:P361+ | wdt:P1269+ | (wdt:P31* / wdt:P279*) ) wd:{q} .
+    ?work wdt:P577 ?datetimes .
+    BIND(xsd:date(?datetimes) AS ?dates)
+  }}
+  GROUP BY ?work ?workLabel
+  ORDER BY DESC(?date)
+  LIMIT 10
+}} AS %content {{
+  INCLUDE %content
+  SERVICE wikibase:label {{ bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }}
+}} ORDER BY DESC(?date)
+"""
+
+def _value(item, field):
+    return item[field]['value'] if field in item else ''
+
+def wb_get_author_latest_articles(q):
+    if not q:
+        return {}
```

Comment:
Why is  a dictionary return here? Shouldn't this be an empty string?

Reply:
Copy/paste issue... no particular reason.

Reply:
I now return an empty string instead in all 3 cases.